public/lessons/www/html.md

```diff
@@ -29,42 +30,61 @@ one HTML file = one page
      house, with two cats in the yard.</p>
     <p>Life used to be <strong>so
      hard</strong>; now everything is
-     easy cause of you.</p>
+     easy because of you.</p>
   </body>
 </html>
 ```
 
 # Elements (aka Tags)
 
-# Attributes
+Lets break down the page structure above. You can see above, all HTML tags are surrounded by angle brackets. You can also see that every tag has what is known as a **closing tag**. These are the tags that have a preceeding forward slash. For the most part, every HTML tag needs to have a closing tag.
 
-# Style vs Layout vs Semantics
+# Examples of Structural Tags (Layout)
 
-The war that has raged inside HTML since the beginning of the WWW.
+|tag|meaning|
+|---|---|
+|\<head>|head (contains metadata)|
+|\<body>|page body|
+|\<div>|division|
+|\<h1>|Heading (level 1)|
+|\<p>|Paragraph containing text|
+|\<span>|Text without a line break|
+|\<img>|Image tag|
 
-# Style Tags
+
+# Examples of Style Tags
 
 |tag|example|
 |---|---|
 |\<b>|<b>bold</b>|
 |\<i>|<i>italic</i>|
 |\<strong>|<strong>strong</strong>|
+|\<em>|<em>emphasis</em>|
+|\<br>|Line break|
+|\<hr>|Horizontal rule (dividing line)|
+|\<width> / \<height>|self-explanatory|
+|\<blockquote>|also self-explanatory|
 |etc|
 
-# Layout Tags
+# Attributes
 
-|tag|meaning|
-|---|---|
-|h1|
-|div|
-|p|
-|hr|
+Attributes further define HTML elements and their purpose. For example, an image tag may have the following attributes:
 
-# Tables
+```<img src="/images/cat-pic.jpg" alt="Picture of a fuzzy cat"```
+
+* src defines where the image file is located.
+* alt is alternative text to be displayed if the image cannot be. 
+* Attributes are not always required. However in the example above, a source is needed for the image to be displayed.
+* Others include style (bad), title (tooltips), href (hyperlink reference)
+* Should always be lowercase
+
+# Style vs Layout vs Semantics
 
-Tables embody the tension between style, layout, and semantics.
+The war that has raged inside HTML since the beginning of the WWW. Some tags exclusively describe _how_ it's contents should be displayed (ex. ```<b>```), where as some _describe_ it's contents (ex. ```<strong>```). Web content isn't just about appearence. It matters how it is intepreted. 
 
-# Floats
+* Semantics are all about meaning - what is the purpose?
+* Not all HTML elements convey meaning - not all HTML elements are semantic (ex. div). 
+* Imagine a blind person using a screen reader - how might they understand the difference between a ```<strong>``` or a ```<b>``` tag?
```

Comment:
Why the triple backticks? I see that it works (though I don't know why) but single-backtick does too.

Reply:
I read in a cheatsheet that codeblocks use triple back ticks. We can change this though. 

"Markdown coverts text with four leading spaces into a code block; with GFM you can
wrap your code with ``` to create a code block without the leading spaces. Add an
optional language identifier and your code will get syntax highlighting."

Reply:
Code blocks are multiline; inline code-font text uses single backticks. (So does Slack FYI.)

Reply:
yep... shoulda known this.... thanks.